### PR TITLE
fix(curriculum): falsy bouncer - do not mutate the original array

### LINKED ISSUE #47622: Falsy Bouncer
### Describe the Issue

The instructions imply that the original array must be modified, while we are fine with a non-mutating approach.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-algorithm-scripting/falsy-bouncer

### Expected behavior

Clearer instructions.

### Screenshots

_No response_

### System

All

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tanmaythakral**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #47622

 My first PR. Let me know if you have any feedbacks. Thanks :D

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting/falsy-bouncer.md [hunk 1]
```diff
@@ -39,7 +39,16 @@ assert.deepEqual(bouncer([false, null, 0, NaN, undefined, '']), []);
...
 ```
+`Orginal array should not be mutated`.
 
+```js
+assert(
+  (function () {
+  let arr = [7, "ate", "", false, null, NaN, 1, 2, undefined];
+  bouncer(arr);
+  return arr.sort().toString() == [7, "ate", "", false, null, NaN, 1, 2, undefined].sort().toString()
+  })());
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting/falsy-bouncer.md:50]:
```suggestion
const arr = ['a', false, 0, 'Naomi'];
bouncer(arr);
assert.deepEqual(arr, ['a', false, 0, 'Naomi'])
```

I don't think we need to be quite so complex here.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting/falsy-bouncer.md (hunk 1 above)
Comment (nayabatir1) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting/falsy-bouncer.md:50]: @naomi-lgbt  Also test description is not added.

Comment (tanmaythakral) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting/falsy-bouncer.md:50]: I tried the suggested change approach earlier, but it doesn't seem to work. By looking at other tests, I came to the conclusion that every code should be in assert for it to work, which is why I've written the code this way,

**Comment by tanmaythakral** [comment]
I've committed the suggested changes.
The test description was already there before.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Should have noticed the test text last night. Apologies.

**Review by naomi-lgbt — DISMISSED** [review]
banger pr, thanks for contributing!

**Review by naomi-lgbt — APPROVED** [review]
Re-approving. Those darn pesky linter errors.

**Comment by tanmaythakral** [comment]
> banger pr, thanks for contributing!

Thank you for guiding me through my first PR. :D

**Review by raisedadead — APPROVED** [review]

**Comment by SaarthakMaini** [comment]
Awesome! I got to learn a lot from this PR, thank you!

**Pull request merged**